# Interval of loop counts to publish ROS message
publish_loop_interval: 10

# Robot root pose
root_pose:
  translation: [0, 0, -0.3]

# Body pose offset
body_pose_offset:
  translation: [0, 0, -0.275]
  rotation: [0, 1.57, 0]

# Number of samples to make bounding box
bbox_sample_num: 1000

# Padding rate of bounding box
bbox_padding_rate: 1.0

# Lower and upper limits of body Yaw angle [deg]
body_yaw_limits: [-45, 45]

# Number of IK trial
ik_trial_num: 3

# Number of IK loop
ik_loop_num: 50

# Threshold of IK [m], [rad]
ik_error_thre: 5e-3

# Constraint space of IK (default is same as template parameter SamplingSpaceType)
ik_constraint_space: SE3

# Body name list for collision avoidance
collision_body_names_list:
  ["BODY", "LARM_LINK2",
  "BODY", "LARM_LINK3",
  "BODY", "LARM_LINK6",
  "CHEST_LINK1", "LARM_LINK2",
  "CHEST_LINK1", "LARM_LINK3",
  "CHEST_LINK1", "LARM_LINK6"]

# Weight of collision task
collision_task_weight: 1e2

####
# The following configurations are not loaded in sampling class but directly loaded in node
####

# Body name list for collision avoidance with door
door_collision_body_names_list:
  ["BODY", "CHEST_LINK1", "LARM_LINK2", "LARM_LINK3"]

# Scale of collision box for door [m]
door_collision_box_scale: [0.05, 0.9, 1.8]
